src/v2/cookbook/practical-use-of-scoped-slots.md

```diff
@@ -0,0 +1,389 @@
+---
+title: Practical use of scoped slots with GoogleMaps
+type: cookbook
+order: 14
+---
+
+## Base Example
+
+There are situations when you want the template inside the slot to be able to access data from the child component that is responsible for rendering the slot content. This is particularly useful when you need freedom in creating custom templates that use the child component's data properties. That is a typical use case for scoped slots.
+
+Imagine a component that configures and prepares an external API to be used in another component, but is not tightly coupled with any specific template. Such a component could then be reused in multiple places rendering different templates but using the same base object with specific API.
+
+We'll create a component (`GoogleMapLoader.vue`) that:
+1. Initializes the [Google Maps API](https://developers.google.com/maps/documentation/javascript/reference/)
+2. Creates `google` and `map` objects
+3. Exposes those objects to the parent component in which the `GoogleMapLoader` is used
+
+Below is an example of how this can be achieved. We will analyze the code piece-by-piece and see what is actually happening in the next section.
+
+Let’s first establish our `GoogleMapLoader.vue` template:
+
+```html
+<template>
+  <div>
+    <div class="google-map" ref="googleMap"></div>
+    <template v-if="Boolean(this.google) && Boolean(this.map)">
+      <slot
+        :google="google"
+        :map="map"
+      />
+    </template>
+  </div>
+</template>
+```
+
+Now, our script needs to pass some props to the component which allows us to set the [Google Maps API](https://developers.google.com/maps/documentation/javascript/reference/) and [Map object](https://developers.google.com/maps/documentation/javascript/reference/map#Map):
+
+```js
+import GoogleMapsApiLoader from "google-maps-api-loader"
+
+export default {
+  props: {
+    mapConfig: Object,
+    apiKey: String,
+  },
+
+  data() {
+    return {
+      google: null,
+      map: null
+    }
+  },
+
+  async mounted() {
+    const googleMapApi = await GoogleMapsApiLoader({
+      apiKey: this.apiKey
+    })
+    this.google = googleMapApi
+    this.initializeMap()
+  },
+
+  methods: {
+    initializeMap() {
+      const mapContainer = this.$refs.googleMap)
```

Comment:
You've got a parenthesis loitering at the end there.

Reply:
Good catch! @mateuszRybczonek do you want to put in another PR for that or if not I can grab it sometime

Reply:
PR with fix here - https://github.com/vuejs/vuejs.org/pull/1848, @alexdilley thanks for spotting that 👍

Reply:
np! I'm selfishly only using merged commit notifications as a way to keep up to speed on new documentation hence not picking up sooner; your entry was an interesting read!

It's maybe not the remit of a recipe in a cookbook, but I, as a reader, was curious to understand why this approach is more appropriate than simply setting up the loader in the parent component, for example, or perhaps more globally in a Vuex store. I think the answer is "reusability", "encapsulation" and "separation of concerns", but I then question how many types of map-related workflows you'd typically find in a project whose requirements couldn't be satisfied by a single, general-purpose map component...splitting concerns and avoiding an unwieldy "god component", maybe? Please appreciate that I may be completely missing the point, for which I apologise!!

There's a lot of buzz around this pattern, with it being a common go-to for those "in the know", and it might be pertinent – with such an example as this as context – to explain _why_ this would be the tool to reach for in this case.

Reply:
np! I'm selfishly only using merged commit notifications as a way to keep up to speed on new documentation hence not picking up sooner; your entry was an interesting read!

It's maybe not the remit of a recipe in a cookbook, but I, as a reader, was curious to understand why this approach is more appropriate than simply setting up the loader in the parent component, for example, or perhaps more globally in a Vuex store. I think the answer is "reusability", "encapsulation" and "separation of concerns", but I then question how many types of map-related workflows you'd typically find in a project whose requirements couldn't be satisfied by a single, general-purpose map component...splitting concerns and avoiding an unwieldy "god component", maybe? Please appreciate that I may be completely missing the point, for which I apologise!!

There's a lot of buzz around this pattern, with it being a common go-to for those "in the know", and it might be pertinent – with such an example as this as context – to explain _why_ this would be the tool to reach for in this case.

Reply:
np! I'm selfishly only using merged commit notifications as a way to keep up to speed on new documentation hence not picking up sooner; your entry was an interesting read!

It's maybe not the remit of a recipe in a cookbook, but I, as a reader, was curious to understand why this approach is more appropriate than simply setting up the loader in the parent component, for example, or perhaps more globally in a Vuex store. I think the answer is "reusability", "encapsulation" and "separation of concerns", but I then question how many types of map-related workflows you'd typically find in a project whose requirements couldn't be satisfied by a single, general-purpose map component...splitting concerns and avoiding an unwieldy "god component", maybe? Please appreciate that I may be completely missing the point, for which I apologise!!

There's a lot of buzz around this pattern, with it being a common go-to for those "in the know", and it might be pertinent – with such an example as this as context – to explain _why_ this would be the tool to reach for in this case.

Reply:
np! I'm selfishly only using merged commit notifications as a way to keep up to speed on new documentation hence not picking up sooner; your entry was an interesting read!

It's maybe not the remit of a recipe in a cookbook, but I, as a reader, was curious to understand why this approach is more appropriate than simply setting up the loader in the parent component, for example, or perhaps more globally in a Vuex store. I think the answer is "reusability", "encapsulation" and "separation of concerns", but I then question how many types of map-related workflows you'd typically find in a project whose requirements couldn't be satisfied by a single, general-purpose map component...splitting concerns and avoiding an unwieldy "god component", maybe? Please appreciate that I may be completely missing the point, for which I apologise!!

There's a lot of buzz around this pattern, with it being a common go-to for those "in the know", and it might be pertinent – with such an example as this as context – to explain _why_ this would be the tool to reach for in this case.